### DISCUSSION TITLE
Is there anyway to set default window size and title bar like UWP in WinUI desktop application?

### DISCUSSION THREAD

**Discussion by sxlllslgh** [microsoft/WindowsAppSDK#950] [Q&A]
In [XAML Controls Gallery](https://github.com/microsoft/Xaml-Controls-Gallery) project which is implemented by UWP, plenty of functions are implemented through UWP API, such as set title bar:
```csharp
public static void UpdateTitleBar(bool isLeftMode) {
    CoreApplication.GetCurrentView().TitleBar.ExtendViewIntoTitleBar = isLeftMode;
    ApplicationViewTitleBar titleBar = ApplicationView.GetForCurrentView().TitleBar;
    if (isLeftMode) {
        NavigationRootPage.Current.NavigationView.PaneDisplayMode = Microsoft.UI.Xaml.Controls.NavigationViewPaneDisplayMode.Auto;
        titleBar.ButtonBackgroundColor = Colors.Transparent;
        titleBar.ButtonInactiveBackgroundColor = Colors.Transparent;
        } else {
        NavigationRootPage.Current.NavigationView.PaneDisplayMode = Microsoft.UI.Xaml.Controls.NavigationViewPaneDisplayMode.Top;
        var userSettings = new UISettings();
        titleBar.ButtonBackgroundColor = userSettings.GetColorValue(UIColorType.Accent);
        titleBar.ButtonInactiveBackgroundColor = userSettings.GetColorValue(UIColorType.Accent);
        }
}
```
Further, in UWP to set window default size, the code could be:
```charp
ApplicationView.PreferredLaunchViewSize = new Size(1600, 900);
ApplicationView.PreferredLaunchWindowingMode = ApplicationViewWindowingMode.PreferredLaunchViewSize;
```
But in WinUI desktop application, I could not find any API like _CoreApplication_ or _ApplicationView_, could it be solved?

What's more, when I used Project Reunion template to create new project, the **Package** project was always created automatically and application could not run without it, is there anyway to remove it and I can run *.exe like traditional programs?

**Comment by maxkatz6** [comment]
I believe you can get win32-handle from WinUI Desktop Window instance. 
With this handle you can use old school win32 APIs to manipulate with windows, including sizes.

**Reply by mdtauk** [reply]
Hopefully Reunion can finally provide a modern API for handling this, to take away the tedious old code for handling this.

Should be good for C# and C++

**Reply by sxlllslgh** [reply]
Thank you for your reply! It seems the only way to approach it now.

**Comment by riverar** [comment]
This is known broken and I hope it gets resolved before 1.0, because the workarounds are absolutely ridiculous. 

https://github.com/microsoft/microsoft-ui-xaml/issues/3689
https://github.com/microsoft/microsoft-ui-xaml/issues/2564
https://github.com/microsoft/microsoft-ui-xaml/issues/2731

**Reply by sxlllslgh** [reply]
Thank you for your reply! Hope modern APIs could be implemented soon.
